ramanujantools/matrix_test.py

```diff
@@ -29,8 +29,17 @@ def test_reduce():
 
 def test_can_call_numerical_subs():
     m = Matrix([[x, 1], [y, 2]])
+
+    # not enough parameters
     assert not m._can_call_numerical_subs({x: 1})
+
+    # some parameters are not integer
     assert not m._can_call_numerical_subs({x: 1, y: y})
+    assert not m._can_call_numerical_subs({x: 1, y: sp.Rational(2, 3)})
```

Comment:
So we want numerical_subs to work only for python built in number classes?
No `sympy` classes like `sp.Float`, `sp.Rational`?
Why?
Here's a block that raises the question
"""
from sympy import symbols

n = symbols('n')
mat = 3 * rn.Matrix([[n**2, sp.Rational(2, 3)], [n + sp.Integer(4), n + sp.Float(5)]])
print("These work:")
print('1:', mat.numerical_subs({n:3}))
print('2', mat.numerical_subs({n:1.4}))
print('3:', mat.numerical_subs({n:sp.Integer(3)}))
print('4:', mat.numerical_subs({n:sp.Float(1.4)}))
print('5:', mat.numerical_subs({n:sp.Rational(1.4, 4.56)}))

print("But `_can_call_numerical_subs` fails on some:")
print(mat._can_call_numerical_subs({n:int(3)}))
print(mat._can_call_numerical_subs({n:float(1.4)}))
print(mat._can_call_numerical_subs({n:sp.Integer(3)}))
print(mat._can_call_numerical_subs({n:sp.Float(1.4)}))
print(mat._can_call_numerical_subs({n:sp.Rational(1.4, 4.56)}))
"""